R/rescale_weights.R

```diff
@@ -2,96 +2,159 @@
+#'   `rescaled_weights_a` "scales the weights so that the new weights sum to the
+#'   cluster sample size" while `rescaled_weights_b` "scales the weights so that
+#'   the new weights sum to the effective cluster size".
+#'
+#'   Regarding the choice between scaling methods A and B, Carle suggests that
+#'   "analysts who wish to discuss point estimates should report results based
+#'   on weighting method A. For analysts more interested in residual
+#'   between-group variance, method B may generally provide the least biased
+#'   estimates". In general, it is recommended to fit a non-weighted model and
+#'   weighted models with both scaling methods and when comparing the models,
+#'   see whether the "inferential decisions converge", to gain confidence in the
+#'   results.
+#'
+#'   Though the bias of scaled weights decreases with increasing group size,
+#'   method A is preferred when insufficient or low group size is a concern.
+#'
+#'   The group ID and probably PSU may be used as random effects (e.g. nested
+#'   design, or group and PSU as varying intercepts), depending on the survey
+#'   design that should be mimicked.
+#'
+#' - `method = "kish"`
+#'
+#'   Rescaling is based on scaling the sample weights so the mean value is 1,
+#'   which means the sum of all weights equals the sample size. Next, the design
+#'   effect (_Kish 1965_) is calculated, which is the mean of the squared weights
+#'   divided by the squared mean of the weights. The scales sample weights are
+#'   then divided by the design effect.
+#'
+#'   Some tests on real-world survey-data suggest that, in comparison to the
+#'   Carle-method, the Kish-method comes closer to estimates from a regular
+#'   survey-design using the **survey** package. Note that these tests are not
+#'   representative and it is recommended to check your results against a
+#'   standard survey-design.
 #'
 #' @references
+#'   - Asparouhov T. (2006). General Multi-Level Modeling with Sampling
+#'   Weights. Communications in Statistics - Theory and Methods 35: 439-460
+#'
 #'   - Carle A.C. (2009). Fitting multilevel models in complex survey data
 #'   with design weights: Recommendations. BMC Medical Research Methodology
 #'   9(49): 1-13
 #'
-#'   - Asparouhov T. (2006). General Multi-Level Modeling with Sampling
-#'   Weights. Communications in Statistics - Theory and Methods 35: 439-460
+#'   - Kish, L. (1965) Survey Sampling. London: Wiley.
+#'
+#' @examplesIf all(insight::check_if_installed(c("lme4", "parameters"), quietly = TRUE))
+#' data(nhanes_sample)
+#' head(rescale_weights(nhanes_sample, "SDMVSTRA", "WTINT2YR"))
+#'
+#' # also works with multiple group-variables
+#' head(rescale_weights(nhanes_sample, c("SDMVSTRA", "SDMVPSU"), "WTINT2YR"))
+#'
+#' # or nested structures.
+#' x <- rescale_weights(
+#'   data = nhanes_sample,
+#'   by = c("SDMVSTRA", "SDMVPSU"),
+#'   probability_weights = "WTINT2YR",
+#'   nest = TRUE
+#' )
+#' head(x)
+#'
+#' \donttest{
```

Comment:
Why is there a `\donttest` here? Isn't the `@examplesIf` tag enough?

Reply:
I thought donttest is for CRAN, so these examples are not run (they take some time to fit)?

Reply:
It's not dontrun